pennylane/interfaces/autograd.py

```diff
@@ -99,5 +106,6 @@ def gradient_product(g):
 
     # define the vector-Jacobian product function for AutogradQNode.evaluate
     autograd.extend.defvjp(AutogradQNode.evaluate, AutogradQNode.QNode_vjp, argnums=[1])
+    qnode._qnode = qnode  # pylint: disable=protected-access
```

Comment:
Could you explain this line? Is this for the case where `qnode_interface is None`?

Reply:
Ah is this part linked to the "possible drawbacks":
> While writing tests for to_autograd(), I noticed that it has a side-effect; it mutates the input QNode in addition to returning a new QNode. I attempted to modify to_autograd() to perform a deep copy to ensure this no longer happens. This worked for simple QNodes, but QNode with tensor observables would not copy correctly --- the output would always be the ground state.

Reply:
Ah, this is more because I want to be consistent with the other interfaces -- that a reference to the 'bare' QNode exists at `self._qnode`.